stdlib/public/Platform/android.modulemap

```diff
@@ -0,0 +1,417 @@
+// The module map for the Android NDK.
+module Android [system] [no_undeclared_includes] {
+  // The features header pulls in cdefs which is
+  // a textual header with C macros.
+  textual header "features.h"
+
+  // The C standard library.
+  module C {
+    link "m"
+
+    // <assert.h>'s use of NDEBUG requires textual inclusion.
+    textual header "assert.h"
+
+    module complex {
```

Comment:
The C standard library headers should all be split into top level modules so that we can get rid of `-fbuiltin-headers-in-system-modules` and `[no_undeclared_includes]`. I would very much prefer that we don't add any more of these mega-modules because we need to take them all apart in order to layer correctly with C++. You'll probably need to do a bit more work to pull out headers used by libc++ and libstdc++. I know the former uses alloca.h and pthread.h and I'm pretty sure sys/types.h too

Reply:
It's android, just libc++, no libstdc++, but yeah, good point.

Reply:
+1

`libc++` definitely has problems with this nested module layout. It needs to be flat at the top level, sadly. The Clang documentation encourages this non-flat design but it won't work for the C library because of interactions between libc++/libstdc++, the C library and the Clang built-in headers.

Reply:
No, that won’t work either, sadly. It’s better than having the implicit includes from one big header, but it’ll still result in circular dependency problems. The individual headers (for the C library specifically) need to be in separate top level modules.

Reply:
FWIW, take a look at the `musl.modulemap` file here, which does what we need it to do: https://github.com/apple/swift/pull/71840/files#diff-56ba06a124c143daf1db4a3fe803f727f5fd8a8bedce4e1307669f2ef113a32b

Reply:
I have a ready module map with multiple split modules, will post it tomorrow.